src/Symfony/Component/Yaml/Inline.php

```diff
@@ -712,8 +717,12 @@ private static function evaluateScalar($scalar, $flags, $references = [])
                         return null;
                     case 0 === strpos($scalar, '!php/const'):
```

Comment:
The legacy tag does not need to be taken into account since they finish with ":", the mapping key is always the new tag.

Reply:
If this works with the legacy tag, this means to me that having an error during parsing when using the new tag is a bug and we should see how to fix that instead of changing the error message.

Reply:
It doesn't work with the legacy tag. If you use the legacy tag as a key, eg `!php/const:MY_CONST: "value"`, then the parsed key is `!php/const` (the new one) because `:` is the key / value separator. So it's impossible to be a in a `$isMappingKey` case and in a legacy tag that ends with `:`.